server/update_branch.go

```diff
@@ -2,60 +2,84 @@ package server
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"net/http"
 	"strings"
 
-	"github.com/mattermost/mattermost-server/v5/mlog"
+	"github.com/mattermost/mattermost-mattermod/model"
 
 	"github.com/google/go-github/v32/github"
 )
 
-func (s *Server) handleUpdateBranch(ctx context.Context, eventIssueComment IssueComment) {
+var (
+	ErrCommenterPermission    error = errors.New("commenter does not have permissions")
+	ErrOrganizationPermission error = errors.New("we don't have permissions")
+	ErrUpdatePullRequest      error = errors.New("could not update pull request")
+)
+
+func (s *Server) handleUpdateBranch(ctx context.Context, eventIssueComment IssueComment) error {
 	prGitHub, _, err := s.GithubClient.PullRequests.Get(ctx,
 		*eventIssueComment.Repository.Owner.Login,
 		*eventIssueComment.Repository.Name,
 		*eventIssueComment.Issue.Number,
 	)
 	if err != nil {
-		mlog.Error("Error getting the latest PR information from github", mlog.Err(err))
-		return
+		return fmt.Errorf("could not get the latest PR information from github: %w", err)
 	}
+
 	pr, err := s.GetPullRequestFromGithub(ctx, prGitHub)
 	if err != nil {
-		mlog.Error("Error Updating the PR in the DB", mlog.Err(err))
-		return
+		return fmt.Errorf("error updating the PR in the DB: %w", err)
 	}
 
 	commenter := eventIssueComment.Comment.User.GetLogin()
+	if err := s.checkUpdatePullRequestPermissions(commenter, pr); err != nil {
+		var msg string
+		switch err {
+		case ErrCommenterPermission:
+			msg = "Looks like you don't have permissions to trigger this command.\n Only available for the PR submitter and org members"
+		case ErrOrganizationPermission:
+			msg = "We don't have permissions to update this PR, please contact the submitter to apply the update."
+		}
+		s.sendGitHubComment(ctx, pr.RepoOwner, pr.RepoName, pr.Number, msg)
+		return err
+	}
+
+	if err := s.updatePullRequest(ctx, pr); err != nil {
```

Comment:
I understand what you are trying to do here. By refactoring to a separate function, we can test this in isolation. But I don't think that's really necessary here. I guess the problem was that the store mocks weren't available, so we couldn't really test `s.GetPullRequestFromGithub`.

I think once my PR goes in we can expand our test coverage by testing that part too, and then bringing this method back inside. We can refactor the `s.sendGitHubComment` part using a defer block and just changing the `msg` variable.

Reply:
Yes, my motivation was exactly what you've said. Actually this function and the permission check can be bundled together. The remaining part, `s.GetPullRequestFromGithub` should be a separate logic simply because it only converts a `github.PullRequest` to `model.PullRequest` for this functions point of view. And It's repeated several times in other places. If you agree I can include those changes into this PR as well.

Reply:
> If you agree I can include those changes into this PR as well.

Up to you. 0/5 on that.

> Actually this function and the permission check can be bundled together. 

By bundled together, you mean in the same method or as a different method? Totally up to you, but I felt we don't really need 2 separate methods for this if we can test everything once we have the store mocks in.